src/resources/auth/google.strategy.ts

```diff
@@ -0,0 +1,65 @@
+import { PassportStrategy } from "@nestjs/passport";
+import { Strategy, VerifyCallback } from "passport-google-oauth20";
+import { Injectable } from "@nestjs/common";
+import { Model } from "mongoose";
+import { InjectModel } from "@nestjs/mongoose";
+import { User } from "./schemas/user.schema";
+import { IMongoUser } from "./user";
+
+@Injectable()
+export class GoogleStrategy extends PassportStrategy(Strategy, "google") {
+  constructor(@InjectModel(User.name) private readonly userModel: Model<User>) {
+    super({
+      clientID: process.env.GOOGLE_CLIENT_ID,
+      clientSecret: process.env.GOOGLE_SECRET,
+      callbackURL: "http://localhost:8080/google/redirect",
+      // callbackURL: "http://localhost:3000",
+      scope: ["email", "profile"],
+      // failureRedirect: "https://uat.design.courtcanva.com",
+      failureRedirect: "http://localhost:3000",
+    });
```

Comment:
This callback URL can use a condition if the environment is developing then back to the callback URL for the development environment. Same for the UAT and production environment.